packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -39,6 +38,36 @@ const stripe = require('stripe').Stripe;
  * @property {Plan['currency']} currency
  */
 
+/**
+ * Determine for two product metadata object's whether the new one
+ * is a valid upgrade for the old one.
+ *
+ * Throws errors if necessary metadata is not present to determine
+ * if its an upgrade.
+ *
+ * @param {AbbrevProduct['product_metadata']} oldMetadata Old product metadata
+ * @param {AbbrevProduct['product_metadata']} newMetadata New product metadata
+ * @returns {boolean} Whether the new product is an upgrade.
+ */
+function validateProductUpgrade(oldMetadata, newMetadata) {
```

Comment:
This is fine as written, but I wonder if it'd be helpful to log something different for the different error cases (or return slightly different errors).

Reply:
FWIW, this function isn't new for this PR - it's just copied over as the last thing used from the subhub API client that this PR otherwise deletes